client/my-sites/term-tree-selector/terms.jsx

```diff
@@ -94,7 +94,8 @@ const TermTreeSelectorList = React.createClass( {
 		const forceUpdate = (
 			! isEqual( prevProps.selected, this.props.selected ) ||
 			prevProps.loading && ! this.props.loading ||
-			( ! prevProps.terms && this.props.terms )
+			( ! prevProps.terms && this.props.terms ) ||
+			( ( prevProps.terms && this.props.terms ) && ( prevProps.terms.length !== this.props.terms.length ) )
```

Comment:
Would `prevProps.terms !== this.props.terms` be sufficient for checking that the terms has changed, since state _shouldn't_ be mutated?